pps_mw_training/utils/scaler.py

```diff
@@ -116,5 +123,99 @@ def from_dict(
                 ]
             ),
             ymin=np.full(len(params), y_min),
+            ymax=np.full(len(params), y_max),
+            apply_log_scale=np.array([p["scale"] == "log" for p in params]),
+        )
+
+
+@dataclass
+class StandardScaler:
+    """Scaler class for Z score scaling"""
+
+    mean: np.ndarray
+    std: np.ndarray
+    apply_log_scale: Optional[np.ndarray] = None
+
+    def _apply_log_scale(self, idx) -> bool:
+        """Check if log scaling should be applied."""
+        if self.apply_log_scale is not None:
+            return self.apply_log_scale[idx]
+        return False
+
+    def apply(
+        self,
+        x: np.ndarray,
+        idx: Optional[int] = None,
+    ) -> np.ndarray:
+        """Apply forward scaling."""
+        if idx is not None:
+            if self._apply_log_scale(idx):
+                x[x <= 0.0] = MIN_VALUE
+                x = np.log(x)
+            return (x - self.mean[idx]) / self.std[idx]
+
+        return np.column_stack(
+            [self.apply(x[:, idx], idx) for idx in range(x.shape[1])]
+        )
+
+    def reverse(
+        self,
+        y: np.ndarray,
+        idx: Optional[int] = None,
+    ) -> np.ndarray:
+        """Apply reversed scaling."""
+        if idx is not None:
+            data = y * self.std + self.mean
+            if self._apply_log_scale(idx):
+                return np.exp(data)
+            return data
+        return np.column_stack(
+            [self.reverse(y[:, idx], idx) for idx in range(y.shape[1])]
+        )
+
+    @staticmethod
+    def get_mean(
+        x: np.ndarray,
+    ) -> float:
+        """Get mean"""
+        return np.float64(np.nanmean(x))
+
+    @staticmethod
+    def get_std(
+        x: np.ndarray,
+    ) -> float:
+        """Get std"""
+        return np.float64(np.nanstd(x))
+
+    @staticmethod
+    def get_zscore_std_mean(param: Dict[str, Union[str, float]], key):
+        try:
+            return param[key]
+        except Exception:
+            return None
+
+    @classmethod
+    def from_dict(
+        cls,
+        params: List[Dict[str, Union[str, float]]],
+    ) -> "StandardScaler":
+        """ "Get scaler object from dict."""
+        return cls(
```

Comment:
Ses comment above. I think it is better if you return something like:
```
return cls(
    mean=np.array([p["mean"] for p in params]),
    std=np.array([p["std"] for p in params]),
    apply_log_scale=np.array(["scale in p and "p["scale"] == "log" for p in params]),
)
```
Possibly within a try and except if you prefer a controlled raise:
```
try:
    return cls(...)
except KeyError:
    raise here something that hints the user that "mean" and "std" must be set for all parameters but it is currently not
```